src/Preferences.js

```diff
@@ -65,6 +65,15 @@ define(function (require, exports, module) {
                 description: Strings.PREF_INDENT_DESC
             },
             inContextMenu: true
+        }, {
+            id: "autohide",
+            type: "boolean",
+            value: true,
+            options: {
+                name: Strings.PREF_AUTOHIDE_NAME,
+                description: Strings.PREF_AUTOHIDE_DESC
+            },
+            inContextMenu: true
```

Comment:
Yes, a button for a toggle is a little bit strange on the panel.
Well, it should work like the button on the main toolbar, highlighted when autohide is enabled.
And it seems that it would not be suitable.

The idea of placing it in the View menu does not like me too much.
Because until now, your extension is self-contained in its own panel and its button in the main toolbar. And personally, I like it that way.

But as you say, maybe it's a reasonable starting point.